### PR TITLE
[rtsan] Introduce halt_on_error flag

### PR DESCRIPTION
**Pull request by cjappl**
This is what we were previously calling "continue mode"

Allow a runtime option to either 

1. Die (halt_on_error=false) -- default behavior
2. Continue

Follow up reviews will allow for de-duplication so users do not drown in info.

### CODE REVIEW

**Comment by cjappl** [comment]
CC Review @davidtrevelyan

File: compiler-rt/lib/rtsan/rtsan.cpp [hunk 1]
```diff
@@ -104,20 +105,18 @@ __rtsan_notify_intercepted_call(const char *func_name) {
...
 SANITIZER_INTERFACE_ATTRIBUTE void
 __rtsan_notify_blocking_call(const char *func_name) {
   __rtsan_ensure_initialized();
   GET_CALLER_PC_BP;
-  ExpectNotRealtime(
-      GetContextForThisThread(),
-      PrintDiagnosticsAndDieAction(
-          {DiagnosticsInfoType::BlockingCall, func_name, pc, bp}));
+  ExpectNotRealtime(GetContextForThisThread(),
+                    DefaultErrorAction({DiagnosticsInfoType::BlockingCall,
```
Comment (cjappl) [inline compiler-rt/lib/rtsan/rtsan.cpp:118]: No functional change here, just renaming this function + clang format messing with whitespace

File: compiler-rt/lib/rtsan/rtsan_flags.inc [hunk 2]
```diff
@@ -16,5 +16,4 @@
 // RTSAN_FLAG(Type, Name, DefaultValue, Description)
 // See COMMON_FLAG in sanitizer_flags.inc for more details.
 
-// Example flag, until we get a real one
-// RTSAN_FLAG(bool, halt_on_error, true, "If true, halt the program on error")
+RTSAN_FLAG(bool, halt_on_error, true, "Exit after first reported error.")
```
Comment (cjappl) [inline compiler-rt/lib/rtsan/rtsan_flags.inc:19]: My preference for this default is to remain true, but I'm open for discussion on the matter.

You can grep for this flag name and see the defaults for the other sanitizers, each takes their own approach

File: compiler-rt/test/rtsan/halt_on_error.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// RUN: %clangxx -fsanitize=realtime %s -o %t
+// RUN: env RTSAN_OPTIONS="halt_on_error=false" %run %t 2>&1 | FileCheck %s
```
Comment (fmayer) [inline compiler-rt/test/rtsan/halt_on_error.cpp:2]: Also add a case with `=true`?

File: compiler-rt/lib/rtsan/rtsan.cpp [hunk 4]
```diff
@@ -43,10 +43,11 @@ static InitializationState GetInitializationState() {
       atomic_load(&rtsan_initialized, memory_order_acquire));
 }
 
-static auto PrintDiagnosticsAndDieAction(DiagnosticsInfo info) {
+static auto DefaultErrorAction(DiagnosticsInfo info) {
```
Comment (fmayer) [inline compiler-rt/lib/rtsan/rtsan.cpp:46]: I don't really like this name. This implies there are other `ErrorAction`s.

Comment (cjappl) [reply compiler-rt/lib/rtsan/rtsan.cpp]: Tried something different, let me know what you think or if you have another suggestion.

This new one matches the name of the parameter ExpectNotRealtime has

**Review by fmayer — APPROVED** [review]
thanks

**Review by davidtrevelyan — APPROVED** [review]

**Pull request merged**